src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -195,31 +199,36 @@ private void ReadConstructorArguments(ref ReadStack state, ref Utf8JsonReader re
                         out _,
                         createExtensionProperty: false);
 
-                    if (state.Current.CtorArgumentState!.FoundProperties == null)
-                    {
-                        state.Current.CtorArgumentState.FoundProperties =
-                            ArrayPool<FoundProperties>.Shared.Rent(Math.Max(1, state.Current.JsonClassInfo.PropertyCache!.Count));
-                    }
-                    else if (state.Current.PropertyIndex - 1 == state.Current.CtorArgumentState.FoundProperties!.Length)
+                    if (jsonPropertyInfo.ShouldDeserialize)
```

Comment:
This `if` check is the major change in this PR which makes sure the JSON property actually maps to an object property, or there is a `JsonExtensionData` property to save the extra property in.